api/envoy/api/v2/route/route.proto

```diff
@@ -325,6 +325,15 @@ message RouteMatch {
   message GrpcRouteMatchOptions {
   }
 
+  message CredentialMatchOptions {
+    // If specified, the route will match against whether or not a certificate is presented.
```

Comment:
Can you make the wording here a bit clearer? I.e. are you talking about the downstream TLS client certificate? If so, I'd rename this, e.g. `TlsContextMatchOptions`. Also, maybe we want to unify with https://github.com/envoyproxy/envoy/blob/878030d3e5fa28db73e2b97fc8c180d24a276c6c/api/envoy/api/v2/route/route.proto#L76 somehow?

Reply:
I've changed this to `TlsContextMatchOptions`.
I'm not sure how to unify with the `TlsRequirementType` because they operate at different levels ?